File: backend/src/main/java/com/example/shop/entity/Address.java
package com.example.shop.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 收货地址表实体（数据字典 2.11 addresses，资料表省略 status）
 * is_default：0否 1是
 */
@Data
@TableName("addresses")
public class Address {

    /** 地址ID */
    @TableId(type = IdType.AUTO)
    private Long id;

    /** 用户ID */
    private Long userId;

    /** 收货人姓名 */
    private String receiver;

    /** 联系电话 */
    private String phone;

    /** 省 */
    private String province;

    /** 市 */
    private String city;

    /** 区/县 */
    private String district;

    /** 详细地址 */
    private String detail;

    /** 是否默认地址：0否 1是 */
    private Integer isDefault;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    /** 软删除：0未删除 1已删除 */
    private Integer deletedFlag;
}

File: frontend-user/src/api/address.js
// 收货地址接口（U-003~U-007，需登录；W3 提供 U-003 供结算页，W5 补齐 U-004~007）
import request from './request'

/** U-003 地址列表（T5 契约：data 为 {list, total}） */
export function getAddresses(config) {
  return request.get('/addresses', config)
}

/**
 * U-004 新增地址（超 20 个返回 2006）
 * @param {Object} data {receiver, phone, province, city, district?, detail, isDefault?}
 * @returns data.id
 */
export function addAddress(data, config) {
  return request.post('/addresses', data, config)
}

/** U-005 编辑地址（校验归属本人 2005） */
export function updateAddress(id, data, config) {
  return request.put(`/addresses/${id}`, data, config)
}

/** U-006 软删除 */
export function deleteAddress(id, config) {
  return request.delete(`/addresses/${id}`, config)
}

/** U-007 设为默认地址 */
export function setDefaultAddress(id, config) {
  return request.put(`/addresses/${id}/default`, null, config)
}
